bootloaders/uart/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+add_executable(uart_boot
+    uart_boot.c
+    )
+
+# pull in common dependencies
+target_link_libraries(uart_boot pico_stdlib hardware_flash)
+
+pico_embed_pt_in_binary(uart_boot ${CMAKE_CURRENT_LIST_DIR}/uart-pt.json)
+pico_set_uf2_family(uart_boot "absolute")
+
+# create map/bin/hex file etc.
+pico_add_extra_outputs(uart_boot)
+
+# add url via pico_set_program_url
+example_auto_set_url(uart_boot)
+
+
+add_executable(uart_binary
+    uart_binary.c
+    )
+
+# pull in common dependencies
+target_link_libraries(uart_binary pico_stdlib)
+
+pico_set_binary_type(uart_binary no_flash)
+pico_package_uf2_output(uart_binary 0x10000000)
```

Comment:
I assume that it's **this** line that puts the `uart_binary` into the partition? Seems quite opaque, and took me quite a while to figure that out :confused: 

Sorry to keep banging on about "magic numbers" but there's another one here - would be nice to at least have a comment indicating what this `0x10000000` means.

Reply:
This doesn't put it in the partition - this just puts it in a packaged SRAM UF2 (datasheet 5.10.3), which when dragged & dropped onto a device will be written to flash rather than SRAM. If you've already dragged & dropped the `uart_boot.uf2` file which contains an embedded partition table, then this `uart_binary.uf2` will be written to the first partition in the Flash, rather than the start of Flash (which is where it would go if there's no partition table on the device).

Ie start with clean device, drag & drop `uart_boot.uf2`, put back in bootsel, then drag & drop `uart_binary.uf2`

Reply:
> Ie start with clean device, drag & drop `uart_boot.uf2`, put back in bootsel, then drag & drop `uart_binary.uf2`

Ohhhh!! That wasn't clear at all - if this is the case then IMHO there definitely needs to be a `README.md` for this example explaining how to use it. My (naive) assumption was that when you built this example you'd just end up with a single UF2 image which contained both the `uart_boot` and `uart_binary` parts in different partitions. (I haven't yet got round to reading up about RP2350 partitions, etc.)